### ISSUE TITLE
Recommended way to check expired status of auto-renewable subscription?

### ISSUE THREAD

**Issue by vanessag** [j3k0/cordova-plugin-purchase#1232]
Hi,

I've been using this plugin in conjunction with the Fovea receipt validator to implement auto-renewable subscriptions in my app and so far it's been great. One thing I'm trying to understand is how do I determine whether the subscription is still active or expired? I would like to display a message in app when the user subscription expires and then remove the benefits.

I see that there is a .expired event but it's not clear and seems inconsistent when this event fires. It seems that if the Fovea validator returns a purchase expired code this event will be appropriately triggered. However, I've noticed that the Fovea validator only returns this code once some time after it expires and then returns a response with the receipt information with the cancelation information.

My question is how to I appropriately determine the status of the subscription? i.e. how do I know all the receipt validation is finished and I have no owned subscriptions?

Any help would be appreciated!

**Comment by vanessag** [comment]
@j3k0 @Dexus Would either of you be able to shed some light on this question?

**Comment by vanessag** [comment]
@HarelM No, sorry. I never did find a good solution to this. The expired event is not very reliable in my experience.

It seems someone else has opened a similar issue and I have commented [there](https://github.com/j3k0/cordova-plugin-purchase/issues/1254)

**Comment by HarelM** [comment]
@vanessag thanks for the info! @j3k0 any updates on this?

**Comment by ahmad-alkhatib** [comment]
Any update on this!

**Comment by jaimyborgman** [comment]
I’m having a similar problem I think. Currently developing locally and while using the sandbox I can subscribe but after expiring or even removing the subscription fully from the App Store manage section the user is still having the subscription. The expired event is not triggered as well and strange enough it is sticking on “owned” as well while the purchase should not be valid anymore
